File: views/hauberg.go
package views

import (
	"html/template"
	"net/http"

	mobiledetect "github.com/Shaked/gomobiledetect"
	"github.com/meehalkoff/loggi"
	"gitlab.com/mvalkov/katzman/lib"
)

func HaubergHandler(w http.ResponseWriter, r *http.Request) {

	tmpl := template.New("hauberg")
	detect := mobiledetect.NewMobileDetect(r, nil)
	if detect.IsMobile() {
		tmpl.ParseFiles(
			"mobile/templates/hauberg.html",
			"mobile/templates/header.html",
			"mobile/templates/footer.html",
			"mobile/templates/top-panel.html",
			"mobile/templates/calc.html",
			"mobile/templates/call.html",
			"mobile/templates/about.html",
			"mobile/templates/firstPromoBlock.html",
			"mobile/templates/creditPromoBlock.html",
			"mobile/templates/lastPromoBlock.html",
			"mobile/templates/how-we-work.html",
		)
	} else {
		tmpl.ParseFiles(
			"templates/header.html",
			"templates/footer.html",
			"templates/top-panel/top-panel.html",
			"templates/top-panel/about.html",
			"templates/top-panel/calc.html",
			"templates/top-panel/call.html",
			"templates/top-panel/main-menu.html",
			"templates/control-panel.html",
			"templates/firstPromoBlock.html",
			"templates/hauberg/index.html",
			"templates/hauberg/summary.html",
			"templates/hauberg/palitra.html",
			"templates/hauberg/kachestva.html",
			"templates/hauberg/schema.html",
			"templates/hauberg/otkos.html",
			"templates/howWeWork.html",
			"templates/hauberg/portfolio.html",
			"templates/shared/video.html",
			"templates/hauberg/promo.html",
			"templates/lastPromoBlock.html",
			"templates/creditPromoBlock.html",
		)
	}

	data := lib.FirstPromoBlock{
		MaterialName: `Фасадная плитка "Технониколь"`,
		ID:           "haubergPrice",
		PictureURL:   "/assets/img/promo/hauberg.png",
		UnitName:     "кв.м.",
		YoutubeID:    "LJAJc8XPbdc",
	}
	if err := tmpl.Execute(w, data); err != nil {
		loggi.Fatal(err)
	}
}

File: assets/js/hauberg.js
var nextArrow = '<div class="control-next-arrow">▶▶</div>';
var prevArrow = '<div class="control-prev-arrow">◀◀</div>';
var maxSlidePosition = 7;

var control = {
    0: {
        id: "first-promo",
        title: "Фиброцементный сайдинг: предложение",
        prev: "",
        next: "Виды покрытий" + nextArrow,
    },
    1: {
        id: "summary",
        title: "Плитка Технониколь: обзор",
        prev: prevArrow + "Предложение",
        next: "Палитра цветов" + nextArrow,
    },
    2: {
        id: "palitra",
        title: "Плитка Технониколь: палитра цветов",
        prev: prevArrow + "Обзор",
        next: "Качества" + nextArrow,
    },
    3: {
        id: "kachestva",
        title: "Плитка Технониколь: качества",
        prev: prevArrow + "Палитра цветов",
        next: "Технология монтажа" + nextArrow,
    },
    4: {
        id: "schema",
        title: "Плитка Технониколь: монтаж",
        prev: prevArrow + "Качества",
        next: "Устройство откосов" + nextArrow,
    },
    5: {
        id: "otkos",
        title: "Плитка Технониколь: откосы",
        prev: prevArrow + "Технология монтажа",
        next: "Портфолио" + nextArrow,
    },
    6: {
        id: "portfolio",
        title: "Плитка Технониколь: портфолио",
        prev: prevArrow + "Устройство откосов",
        next: "Видео обзор" + nextArrow,
    },
    7: {
        id: "video",
        title: "Плитка Технониколь: видео обзор",
        prev: prevArrow + "Портфолио",
        next: "Предложение" + nextArrow,
    },
}

var maxPortfolioPosition = 7;

var portfolio = {
    0: 'url(/assets/img/hauberg/portfolio/0.jpg)',
    1: 'url(/assets/img/hauberg/portfolio/1.jpg)',
    2: 'url(/assets/img/hauberg/portfolio/2.jpg)',
    3: 'url(/assets/img/hauberg/portfolio/3.jpg)',
    4: 'url(/assets/img/hauberg/portfolio/4.jpg)',
    5: 'url(/assets/img/hauberg/portfolio/5.jpg)',
    6: 'url(/assets/img/hauberg/portfolio/6.jpg)',
    7: 'url(/assets/img/hauberg/portfolio/7.jpg)',
    8: 'url(/assets/img/hauberg/portfolio/8.jpg)',
}